chainer/functions/pooling/average_pooling_2d.py

```diff
@@ -21,6 +26,30 @@ def forward_cpu(self, x):
         y = col.mean(axis=(2, 3))
         return y,
 
+    def _forward_ideep(self, x):
+        self._in_shape = x[0].shape
+        self._in_dtype = x[0].dtype
+
+        n, c, h, w = x[0].shape
+        y_h = conv.get_conv_outsize(
+            h, self.kh, self.sy, self.ph, self.cover_all)
+        assert y_h > 0, 'Height in the output should be positive.'
+        y_w = conv.get_conv_outsize(
+            w, self.kw, self.sx, self.pw, self.cover_all)
+        assert y_w > 0, 'Width in the output should be positive.'
+        self.pd = self.sy * (y_h - 1) + self.kh - h - self.ph
+        self.pr = self.sx * (y_w - 1) + self.kw - w - self.pw
```

Comment:
Is it intentional to keep `pd` and `pr` as an instance attribute?

Reply:
I don't think so. Fixed.